### PR TITLE
Add Gemini Driver

### LINKED ISSUE #4: Gemini text completions
## Description

Create a new driver for Gemini and add it to the default config

https://ai.google.dev/gemini-api/docs

### PR DESCRIPTION
**Pull request by yoavf**
This adds a Gemini driver with support for:
- System messages
- Function calling

Fixes #4

### CODE REVIEW

File: src/Drivers/Gemini/Client.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Facades\Http;
+
+class Client
+{
+    private readonly PendingRequest $client;
+
+    private const string BASE_URL = 'https://generativelanguage.googleapis.com/v1beta';
```
Comment (pushpak1300) [inline src/Drivers/Gemini/Client.php:13]: don't you think this should be config value ?

Comment (yoavf) [reply src/Drivers/Gemini/Client.php:13]: Yep, you're right - I'll update.

File: config/prism.php [hunk 2]
```diff
@@ -15,6 +15,11 @@
             'api_key' => env('ANTHROPIC_API_KEY'),
             'version' => env('ANTHROPIC_API_VERSION', '2023-06-01'),
         ],
+        'gemini' => [
```
[the remark this answers is not in the document]
Comment (pushpak1300) [reply config/prism.php:18]: https://python.langchain.com/docs/integrations/platforms/google/

https://sdk.vercel.ai/providers/ai-sdk-providers/google-generative-ai

[the remark this answers is not in the document]
Comment (yoavf) [reply config/prism.php:18]: "Google" felt a bit too generic to me, but I don't have a super strong opinion about this. I'll leave it to @sixlive to make the call.

[the remark this answers is not in the document]
Comment (sixlive) [reply config/prism.php:18]: I'm honestly a little torn, so far we've used organization names as driver names, e.x. Anthropic and OpenAI. This library takes heavy inspiration from the Vercel AI SDK and it looks like they also call their provider google. 

I lets go with Google for now.

**Comment by sixlive** [comment]
Give me a day or so to review this and let it simmer.

File: config/prism.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (yoavf) [reply config/prism.php]: Done :)

**Comment by petervandijck** [comment]
Is this still being worked on? I'm waiting to start using prism once it supports gemini (and streaming chat)

**Comment by sixlive** [comment]
I churned a whole bunch with recent refactors. Worst case I'll make sure to get a Gemini provider knocked out this weekend.

**Comment by sixlive** [comment]
@pushpak1300 is going to take another pass at this after we chruned a bit on some standards and functionality. Appreciate the initial implementation!

**Pull request closed without merging**